### PR TITLE
dhclient: T3832: Add hexadecimal format for client-id

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to use hexadecimal format for DHCP client option client-id.
The value of this option should be without double quotes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3832

## How to test
```
set interfaces ethernet eth2 address 'dhcp'
set interfaces ethernet eth2 dhcp-options client-id '23:33:bb:aa:dd:2d:33:2f:ff'
```
Expected to see client-id without quotes:
```
vyos@r1-roll# cat /var/lib/dhcp/dhclient_eth2.conf
### Autogenerated by interface.py ###

option rfc3442-classless-static-routes code 121 = array of unsigned integer 8;
timeout 60;
retry 300;

interface "eth2" {
    send host-name "r1-roll";
    send dhcp-client-identifier 23:33:bb:aa:dd:2d:33:2f:ff

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
Why not always move to decimal output and detect on demand if colons are present or not and adjust the string? Ne need for the user to take any action at all?

**Comment by sever-sever** [comment]
> Why not always move to decimal output and detect on demand if colons are present or not and adjust the string? Ne need for the user to take any action at all?

The first idea was to detect colons.
But if client id will be `MyOffice:PC1`

**Comment by c-po** [comment]
Count the colons and switch to hex if count >=4?

**Comment by sever-sever** [comment]
> Count the colons and switch to hex if count >=4?

Done

File: data/templates/dhcp-client/ipv4.tmpl [hunk 1]
```diff
@@ -7,7 +7,7 @@ retry 300;
 interface "{{ ifname }}" {
     send host-name "{{ dhcp_options.host_name }}";
 {% if dhcp_options.client_id is defined and dhcp_options.client_id is not none %}
-    send dhcp-client-identifier "{{ dhcp_options.client_id }}";
+    send dhcp-client-identifier {{ dhcp_options.client_id if ':' in dhcp_options.client_id and dhcp_options.client_id.split(':')|length >= 5 else "\"{}\"".format(dhcp_options.client_id) }};
```
Comment (c-po) [inline data/templates/dhcp-client/ipv4.tmpl]: Why not make this even more understandable by writing:

```
{%   set client_id = dhcp_options.client_id %}
{%   if dhcp_options.client_id.split(':') | length >= 5 %}
{#   Use HEX representation of client-id as it is send in MAC-address style using hex characters #}
{%     set client_id = "{" + dhcp_options.client_id + "}" %}
{%   endif %}
    send dhcp-client-identifier {{ client_id  }};
```

**Review by c-po — CHANGES_REQUESTED** [review]

**Pull request merged**